flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/sink/KafkaWriterITCase.java

```diff
@@ -147,13 +147,18 @@ public void testIncreasingRecordBasedCounters() throws Exception {
         try (final KafkaWriter<Integer> writer =
                 createWriterWithConfiguration(
                         getKafkaClientConfiguration(), DeliveryGuarantee.NONE, metricGroup)) {
-            final Counter numBytesOut = operatorIOMetricGroup.getNumBytesOutCounter();
-            final Counter numRecordsOut = operatorIOMetricGroup.getNumRecordsOutCounter();
-            assertEquals(numBytesOut.getCount(), 0L);
+            final Counter numBytesSend = metricGroup.getNumBytesSendCounter();
+            final Counter numRecordsSend = metricGroup.getNumRecordsSendCounter();
+            final Counter numRecordsWrittenErrors = metricGroup.getNumRecordsOutErrorsCounter();
+            assertEquals(numBytesSend.getCount(), 0L);
```

Comment:
Please use assertJ for new assertions `assertThat(...).isEqualTo()`

Reply:
This is a JUnit 4 test. It make more sense to do the migration in the separate thread. WDYT?

Reply:
I think the test is already junit 5 but yes probably it makes sense to switch the whole class to use assertJ as separate commit. If you want to do it on this PR fine but also okay if you keep the current assertions.

Reply:
you are right. I will do it in new commit in this PR.